word2vec_test.go

```diff
@@ -4,10 +4,41 @@ import (
 	"bytes"
 	"encoding/binary"
 	"fmt"
+	"math/rand"
 	"reflect"
 	"testing"
+
+	"github.com/ziutek/blas"
+	"gonum.org/v1/gonum/blas/blas32"
 )
 
+func data(n int) []float32 {
+	rand.Seed(10)
+	data := []float32{}
+	for index := 0; index < n; index++ {
+		data = append(data, rand.Float32())
+	}
+	return data
+}
+
+func BenchmarkDotGonumFloat32(b *testing.B) {
+	d := data(300)
```

Comment:
Can you turn this into multiple benchmarks (ideally using sub-benchmarks) that test for different sizes of vectors: 100, 150, 200, 250, 300.  It's then easy to add more dimensions to check.